clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroComputeRequestBuilder.java

```diff
@@ -364,7 +363,13 @@ public void onCompletion(Optional<Exception> exception) {
   }
 
   @Override
-  public void streamingExecute(Set<K> keys, StreamingCallback<K, GenericRecord> callback) throws VeniceClientException {
+  public void streamingExecute(Set<K> keys, StreamingCallback<K, ComputeGenericRecord> callback)
+      throws VeniceClientException {
+    if (executed) {
```

Comment:
If we want to add a safe-guard against this, we need to make the methods either synchronized or create a critical section for `executed`, right?

Reply:
Even we don't add this `executed` check, it should be fine since the response is either returned as an object or trigger some user-supplied callback, right?

Keeping the current loose constraint might be fine since we don't prevent the invocation of `setter` methods when calling `execute`.

If we really want to have this new semantics, maybe we should call it out in the javadoc of this class to mention this is not thread-safe.

Reply:
The class implementation is not thread-safe, and doesn't need to be. The check is to prevent old operations (e.g. `project()`) from carrying over to the subsequent `execute()` call.